test/harry/main/org/apache/cassandra/harry/model/ASTSingleTableModel.java

```diff
@@ -0,0 +1,1289 @@
+        if (ctx.unmatchable)
+        {
+            primaryKeys = Collections.emptyList();
+        }
+        else if (ctx.eq.keySet().containsAll(factory.pkPositions))
+        {
+            primaryKeys = findByPartitionEq(ctx);
+        }
+        else if (ctx.token != null)
+        {
+            primaryKeys = findKeysByToken(ctx);
+        }
+        else if (ctx.tokenLowerBound != null || ctx.tokenUpperBound != null)
+        {
+            primaryKeys = findKeysByTokenSearch(ctx);
+        }
+        else
+        {
+            primaryKeys = search(ctx);
+        }
+        //TODO (correctness): now that we have the rows we need to handle the selections/aggregation/limit/group-by/etc.
+        return new SelectResult(getRowsAsByteBuffer(primaryKeys), ctx.unordered);
+    }
+
+    public ByteBuffer[][] getRowsAsByteBuffer(List<BytesPartitionState.PrimaryKey> primaryKeys)
+    {
+        ByteBuffer[][] rows = new ByteBuffer[primaryKeys.size()][];
+        int idx = 0;
+        for (BytesPartitionState.PrimaryKey pk : primaryKeys)
+        {
+            BytesPartitionState partition = partitions.get(pk.partition);
+            BytesPartitionState.Row row = partition.get(pk.clustering);
+            rows[idx++] = getRowAsByteBuffer(partition, row);
+        }
+        return rows;
+    }
+
+    public ByteBuffer[][] getRowsAsByteBufferFromPartitions(List<BytesPartitionState> ps)
+    {
+        List<ByteBuffer[]> rows = new ArrayList<>();
+        for (BytesPartitionState p : ps)
+        {
+            for (BytesPartitionState.Row row : p.rows())
+                rows.add(getRowAsByteBuffer(p, row));
+        }
+        return rows.toArray(ByteBuffer[][]::new);
+    }
+
+    private ByteBuffer[] getRowAsByteBuffer(BytesPartitionState partition, @Nullable BytesPartitionState.Row row)
+    {
+        Clustering<ByteBuffer> pd = partition.key;
+        BytesPartitionState.Row staticRow = partition.staticRow();
+        ByteBuffer[] bbs = new ByteBuffer[factory.selectOrder.size()];
+        for (Symbol col : factory.pkPositions)
+            bbs[factory.selectOrder.indexOf(col)] = pd.bufferAt(factory.pkPositions.indexOf(col));
+        for (Symbol col : factory.staticPositions)
+            bbs[factory.selectOrder.indexOf(col)] = staticRow.get(col);
+        if (row != null)
+        {
+            for (Symbol col : factory.ckPositions)
+                bbs[factory.selectOrder.indexOf(col)] = row.clustering.bufferAt(factory.ckPositions.indexOf(col));
+            for (Symbol col : factory.regularPositions)
+                bbs[factory.selectOrder.indexOf(col)] = row.get(col);
+        }
+        return bbs;
+    }
+
+    private LookupContext context(Select select)
+    {
+        if (select.where.isEmpty())
+            throw new IllegalArgumentException("Select without a where clause is currently unsupported");
+        LookupContext ctx = new LookupContext(select);
+        context(ctx, select.where.get());
```

Comment:
Move this and the next line into the `LookupContext` constructor?

Reply:
done